File: codigo/backend/sap-ecc-emulator/src/models/loja.py
from db import db

class Loja(db.Model):
    __tablename__ = 'lojas'

    id = db.Column(db.Integer, primary_key=True)
    nome = db.Column(db.String(255), nullable=False)
    endereco = db.Column(db.String(255), nullable=False)

    def to_dict(self):
        return {
            'id': self.id,
            'nome': self.nome,
            'endereco': self.endereco
        }

    def __repr__(self):
        return f'<Loja {self.nome}>'

File: codigo/backend/go-backend/models/loja.go
package models
import "fmt"


type Loja struct {
    ID       uint   `gorm:"primaryKey"`
    Nome     string `gorm:"size:255;not null"`
    Endereco string `gorm:"size:255;not null"`
}


func (Loja) TableName() string {
    return "lojas" // Nome da tabela no banco de dados
}

func (l *Loja) ToMap() map[string]interface{} {
    return map[string]interface{}{
        "id":       l.ID,
        "nome":     l.Nome,
        "endereco": l.Endereco,
    }
}

func (l *Loja) String() string {
    return fmt.Sprintf("<Loja %s>", l.Nome)
}
